cip/1.accepted/CIP2015-05-13-EXISTS.adoc

```diff
@@ -4,102 +4,185 @@
-           ;
+==== Grammar
 
-property exists = "exists", "(", expression ")"
-                ;
+[source,bnf]
+----
+<Atom> ::=
+    ...
+  | <RelationshipsPattern>
+  | <ExistentialSubquery>
 
-subquery exists = "EXISTS", "{", read only clause, { read only clause }, "}" ;
+<ExistentialSubquery> ::=
+    <SimpleExistentialSubquery>
+  | <FullExistentialSubquery>
 
-simple subquery exists = "EXISTS", "{", simple match, "}" ;
+<SimpleExistentialSubquery> ::=
+  "EXISTS" "{" <SimpleMatch> "}"
 
-simple match = pattern, { ",", pattern }, [ "WHERE", predicate ] ;
+<FullExistentialSubquery> ::=
+  "EXISTS" "{" <RegularQuery> "}"
 
-read only clause = match
-                 | unwind
-                 | with
-                 ;
+<SimpleMatch> ::=
+  <Pattern> [ <Where> ]
 ----
 
+Note that the openCypher grammar does not list <SimpleExistentialSubquery>, <SimpleMatch>, and <FullExistentialSubquery> as separate productions but represents them inline of https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=ExistentialSubquery[<ExistentialSubquery>].
+
+==== Syntax Rules
+
+* A https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=RelationshipsPattern[<RelationshipsPattern>] shall only be contained in sites whose expected type is exactly boolean. Specifically, a https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=RelationshipsPattern[<RelationshipsPattern>] shall only be simply contained in
+** <Where>, used in
+*** https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=Match[<Match>]
+*** https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=With[<With>]
+*** https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=YieldItems[<YieldItems>]
+*** https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=FilterExpression[<FilterExpression>], used in
+**** `ALL`, `ANY`, `NONE`, `SINGLE`
+**** https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=ListComprehension[<ListComprehension>]
+*** https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=PatternComprehension[<PatternComprehension>]
+** https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=CaseAlternatice[<CaseAlternative>]
+* The https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=RegularQuery[<RegularQuery>] contained in a <FullExistentialSubquery> shall not contain any https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=UpdatingClause[<UpdatingClause>] nor procedure or function calls with any side effects
```

Comment:
question on the definition of "syntax rule": I understand that updating clauses are syntactically identifiable, but procs and functions _with side effects_ are not. Do these still belong under this heading?

Reply:
That is debatable either way. I do not have strong opinions on it. My assumption is that the implementation would statically know if something may cause side effects, which suggests a slight refined wording though:

```suggestion
* The https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=RegularQuery[<RegularQuery>] contained in a <FullExistentialSubquery> shall not contain any https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=UpdatingClause[<UpdatingClause>] nor procedure or function calls that are not known to be free of side effects.
```